flake.nix

```diff
@@ -1,64 +1,62 @@
-      inputs.emanote.follows =
-        "plutarch/haskell-nix/nixpkgs-unstable";
-      inputs.nixpkgs.follows =
-        "plutarch/haskell-nix/nixpkgs-unstable";
+    liqwid-nix = {
+      ################################################### change this before commit
+      url = "github:Liqwid-Labs/liqwid-nix/liqwid-nix-2.0";
+      inputs.nixpkgs-latest.follows = "nixpkgs-latest";
     };
 
+    liqwid-plutarch-extra.url = "github:Liqwid-Labs/liqwid-plutarch-extra/emiflake/liqwid-nix-2.0";
+    plutarch-quickcheck.url = "github:Liqwid-Labs/plutarch-quickcheck/emiflake/liqwid-nix-2.0";
+    plutarch-numeric.url = "github:Liqwid-Labs/plutarch-numeric/emiflake/liqwid-nix-2.0";
     ply.url = "github:mlabs-haskell/ply?ref=master";
-
-    liqwid-nix.url = "github:Liqwid-Labs/liqwid-nix";
-    liqwid-plutarch-extra.url = "github:Liqwid-Labs/liqwid-plutarch-extra";
-    plutarch-numeric.url =
-      "github:liqwid-labs/plutarch-numeric?ref=main";
-
-
-    # liqwid-script-import
-
-    ctl = {
-      type = "github";
-      owner = "Plutonomicon";
-      repo = "cardano-transaction-lib";
-      rev = "39fc516c7c837401427b46637dacd9c7ada6274d";
-    };
   };
 
-  outputs = inputs@{ self, nixpkgs, ... }:
-    let
-      defaultSystems =
-        [ "x86_64-linux" "x86_64-darwin" "aarch64-linux" "aarch64-darwin" ];
-      perSystem = nixpkgs.lib.genAttrs defaultSystems;
-    in
-    rec {
-      liqwid-script-import = {
-        flake = (import ./liqwid-script-import/flake.nix) { inherit inputs liqwid-script-export; };
-      };
-      liqwid-script-export = {
-        flake = (import ./liqwid-script-export/flake.nix) { inherit inputs; };
-      };
-
-      devShells = perSystem (system: rec {
-        default = liqwid-script-export;
-        liqwid-script-import = self.liqwid-script-import.flake.devShell.${system};
-        liqwid-script-export = self.liqwid-script-export.flake.devShell.${system};
-      });
+  outputs = { self, liqwid-nix, flake-parts, ... }:
+    flake-parts.lib.mkFlake { inherit self; } {
+      imports = liqwid-nix.allModules ++ [
+        ({ self, ... }:
+          {
+            perSystem = { config, pkgs', self', inputs, system, ... }:
+              let
+                pkgs = import self.inputs.nixpkgs {
+                  inherit system;
+                };
+              in
+              {
+                onchain.default = {
+                  src = ./liqwid-script-export;
+                  ghc.version = "ghc925";
+                  shell = { };
+                  enableBuildChecks = true;
+                  extraHackageDeps = [
+                    "${self.inputs.plutarch-quickcheck}"
+                    "${self.inputs.plutarch-numeric}"
+                    "${self.inputs.liqwid-plutarch-extra}"
+                    "${self.inputs.ply}/ply-core"
+                    "${self.inputs.ply}/ply-plutarch"
+                  ];
+                };
+                ci.required = [ "all_onchain" ];
+              };
+          })
```

Comment:
It's nicer to import this module from a separate file, especially once you add the off-chain module.

Reply:
@emiflake and I talked about having separate `flake-module.nix`s. We both think that it is kind of useless for repos with single package, but I'm not sure if thats same for multi package repo. 

Personally, I think it's okay to just have one `flake.nix` because it's really not much to have both on one file.